### PR TITLE
New recipe galaxy-w4m-r-classfilter

### PR DESCRIPTION
**Pull request by eschen42**
* [x] I have read the guidelines above.
* [x] This PR adds a new recipe.
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

Please review and provide suggestions for improvement if appropriate.

This is a new recipe for an R package that will implement processing for a new tool for the Workflow4Metabolomics flavor of Galaxy.  My intent is to add just two files:
  - recipes/galaxy-w4m-r-classfilter/build.sh
  - recipes/galaxy-w4m-r-classfilter/meta.yaml

This is my first pull request (ever), so, if there are more files than these two, that's a problem that I didn't detect!

The travis build failed for the commit that added them (which is what I believe is the basis of this pull request); however, [in that build](https://travis-ci.org/eschen42/bioconda-recipes/jobs/228312460), **this recipe succeeded**, viz:
> 14:27:31 BIOCONDA INFO BUILD START recipes/galaxy-w4m-r-classfilter, env: CONDA_BOOST=1.61;CONDA_GMP=5.1;CONDA_GSL=1.16;CONDA_HDF5=1.8.17;CONDA_HTSLIB=1.4;CONDA_NCURSES=5.9;CONDA_NPY=110;CONDA_PERL=5.22.0;CONDA_PY=27;CONDA_R=3.3.1
>
> 14:28:53 BIOCONDA INFO BUILD SUCCESS /anaconda/conda-bld/linux-64/w4mclassfilter-0.98.0-0.98.0.tar.bz2, CONDA_BOOST=1.61;CONDA_GMP=5.1;CONDA_GSL=1.16;CONDA_HDF5=1.8.17;CONDA_HTSLIB=1.4;CONDA_NCURSES=5.9;CONDA_NPY=110;CONDA_PERL=5.22.0;CONDA_PY=27;CONDA_R=3.3.1
>
> 14:28:53 BIOCONDA INFO TEST START via mulled-build recipes/galaxy-w4m-r-classfilter, CONDA_BOOST=1.61;CONDA_GMP=5.1;CONDA_GSL=1.16;CONDA_HDF5=1.8.17;CONDA_HTSLIB=1.4;CONDA_NCURSES=5.9;CONDA_NPY=110;CONDA_PERL=5.22.0;CONDA_PY=27;CONDA_R=3.3.1
>
> updating index in: /anaconda/conda-bld/linux-64
>
> 14:30:32 BIOCONDA INFO TEST SUCCESS recipes/galaxy-w4m-r-classfilter, CONDA_BOOST=1.61;CONDA_GMP=5.1;CONDA_GSL=1.16;CONDA_HDF5=1.8.17;CONDA_HTSLIB=1.4;CONDA_NCURSES=5.9;CONDA_NPY=110;CONDA_PERL=5.22.0;CONDA_PY=27;CONDA_R=3.3.1

The next travis build succeeded, but did not build this recipe. ugh.

### CODE REVIEW

**Comment by eschen42** [comment]
Now that I have rendered my pull request as a diff using the URL
[https://patch-diff.githubusercontent.com/raw/bioconda/bioconda-recipes/pull/4594.diff](https://patch-diff.githubusercontent.com/raw/bioconda/bioconda-recipes/pull/4594.diff), I have greater confidence that the changes included are the changes intended.
:)

File: recipes/galaxy-w4m-r-classfilter/build.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
+#!/bin/bash
+
+# R refuses to build packages that mark themselves as
```
Comment (bgruening) [inline recipes/galaxy-w4m-r-classfilter/build.sh]: Can you please remove the comments here and the first empty line. First line should be the shebang.

File: recipes/galaxy-w4m-r-classfilter/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  md5: 2f3ef13570c3d033bd3e24ada5d52e71
+build:
+  number: 0
+  string: 0.98.0
+  rpaths:
+    - lib/R/lib/
+    - lib/
+requirements:
+  build:
+    - r
```
Comment (bgruening) [inline recipes/galaxy-w4m-r-classfilter/meta.yaml]: r-base instead or `r`

**Comment by eschen42** [comment]
@bgruening pls see commit https://github.com/bioconda/bioconda-recipes/pull/4594/commits/53fb8b37f30f1326ae689c1645af6cf79c7aa51e 

To be honest, I don't understand how changes that I pushed to https://github.com/eschen42/bioconda-recipes resulted in a commit to https://github.com/bioconda/bioconda-recipes, but at least it isn't something that I regret at the end of the day.

**Comment by bgruening** [comment]
@eschen42 everything is normal as far as I can see and this is a great first contribution! Thanks a bunch. When you create a Pull request. You request us (BioConda) to pull changes from your branch (a specific one). If you push to this branch (on your local repo) and this Pull request is still open you still ask us to pull the branch with the recent changes. Does this make sense?

Thanks again!

**Comment by eschen42** [comment]
Great! Thank you for the explanation to help a newbie get his footing.

**Comment by eschen42** [comment]
@bgruening I seem to have done something unexpected.
With good intentions but perhaps bad judgment, I named my recipe differently from my package.
My R package is named w4mclassfilter, and that is the name that I used for the package in the Bioconda recipe because I was interpreting this as the name of the payload package rather than the name of a Bioconda payload-delivery-package.  So, **what to do**?

- move and rename my recipe from 'galaxy-w4m-r-classfilter' to 'w4mclassfilter' ?
- change my package name in meta.yaml to 'galaxy-w4m-r-classfilter' ?
- something that I haven't thought of ?

I am leaning toward the first option.

Thank you for your insight and assistance.

**Comment by bgruening** [comment]
@eschen42 just rename the folder, so option one and create a PR. This should be fine.
Thanks!

**Comment by eschen42** [comment]
@bgruening Excellent! And thanks for the super-fast reply!

**Pull request merged**